app/services/package.scala

```diff
@@ -0,0 +1,59 @@
+import com.gu.identity.play.IdMinimalUser
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+
+package object services {
+
+  case class UserId(id:String) {
+    override def toString = id
+  }
+
+  object UserId {
+    val paramName = "user-id"
+
+    implicit val jsonReader = new Reads[UserId] {
+      override def reads(json: JsValue): JsResult[UserId] = json match {
+        case JsString(id) => JsSuccess(UserId(id))
+        case _            => JsError("Cannot parse UserId")
+      }
+    }
+  }
+
+  case class IdentityToken(token: String) {
+    override def toString = token
+  }
+
+  object IdentityToken {
+    val paramName = "identity-token"
+
+    implicit val jsonReader = new Reads[IdentityToken] {
+      override def reads(json: JsValue): JsResult[IdentityToken] = json match {
+        case JsString(id) => JsSuccess(IdentityToken(id))
+        case _            => JsError("Cannot parse IdentityToken")
+      }
+    }
+  }
+
+  sealed trait UserIdData {
+    def id: UserId
+  }
+
+  case class MinimalIdUser(user: IdMinimalUser) extends UserIdData {
```

Comment:
Okay I see - I didn't spot where it was being used in the Controller. I think this is okay but we need to use IdMinimalUser when implementing `test-users` and I'm a bit concerned we're going to get ourselves confused with the type of User. @rtyley what do you think?